src/Forms/ValidatorList.php

```diff
@@ -0,0 +1,206 @@
+            $validator->resetResult();
+
+            // This Validator has been disabled, so skip it
+            if (!$validator->getEnabled()) {
+                continue;
+            }
+
+            // Run validation, and exit early if it's valid
+            if ($validator->php($data)) {
+                continue;
+            }
+
+            // Validation result was invalid. Combine our ValidationResult messages
+            $this->getResult()->combineAnd($validator->getResult());
+        }
+
+        return $this->result;
+    }
+
+    /**
+     * Note: The existing implementations for the php() method (@see RequiredFields) does not check whether the
+     * Validator is enabled or not, and it also does not reset the validation result - so, neither does this.
+     *
+     * @param array $data
+     * @return bool
+     */
+    public function php($data)
+    {
+        foreach ($this->getValidators() as $validator) {
+            // Run validation, and exit early if it's valid
+            if ($validator->php($data)) {
+                continue;
+            }
+
+            // Validation result was invalid. Combine our ValidationResult messages
+            $this->getResult()->combineAnd($validator->getResult());
+        }
+
+        // After collating results, return whether or not everything was valid
+        return $this->getResult()->isValid();
+    }
+
+    /**
+     * Returns whether the field in question is required. This will usually display '*' next to the
+     * field.
+     *
+     * @param string $fieldName
+     *
+     * @return bool
+     */
+    public function fieldIsRequired($fieldName)
+    {
+        foreach ($this->getValidators() as $validator) {
+            if ($validator->fieldIsRequired($fieldName)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @return array|Validator[]
+     */
+    public function getValidators(): array
+    {
+        return $this->validators;
+    }
+
+    /**
+     * @param string $className
+     * @return array|Validator[]
+     */
+    public function getValidatorsByType(string $className): array
+    {
+        $validators = [];
+
+        foreach ($this->getValidators() as $key => $validator) {
+            if (!$validator instanceof $className) {
+                continue;
+            }
+
+            $validators[$key] = $validator;
+        }
+
+        return $validators;
+    }
+
+    /**
+     * @param string $className
+     * @return ValidatorList
+     */
+    public function removeValidatorsByType(string $className): ValidatorList
+    {
+        foreach ($this->getValidatorsByType($className) as $key => $validator) {
+            $this->removeValidatorByKey($key);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param int $key
+     * @return ValidatorList
+     */
+    public function removeValidatorByKey(int $key): ValidatorList
```

Comment:
How confident are we that this is the right public API to add?

Reply:
Happy to remove it. If it's a feature that folks want in the future, they'll ask for it.

Reply:
I agree - it feels a bit odd, whereas `removeValidatorsByType` makes a lot of sense, and matches the GridFieldConfig equivalent nicely

Reply:
Maybe protected and put `@internal` in the docblock so people know we might change it in the future.

Reply:
Sounds good. Thanks.